src/SplitTestsByGroups.php

```diff
@@ -84,17 +112,83 @@ public function run()
         $this->printTaskInfo("Processing ".count($tests)." tests");
         // splitting tests by groups
         foreach ($tests as $test) {
-            $groups[($i % $this->numGroups) + 1][] = \Codeception\Test\Descriptor::getTestFullName($test);
-            $i++;
+            if (empty($this->filterGroup) || $this->matchesToFilterGroup($test)) {
+                $fullname = str_replace('\\', '/', \Codeception\Test\Descriptor::getTestFullName($test));
+                $groups[($i % $this->numGroups) + 1][] = $fullname;
+                $i++;
+            }
         }
 
+        $this->printTaskInfo(
+            sprintf(
+                "%s test%s match%s the criterias. They will be divided in up to %s groups.",
+                $i,
+                ($i == 1 ? '' : 's'),
+                ($i == 1 ? 'es' : ''),
+                $this->numGroups
+            )
+        );
+
+
         // saving group files
         foreach ($groups as $i => $tests) {
             $filename = $this->saveTo . $i;
-            $this->printTaskInfo("Writing $filename");
+            $this->printTaskInfo(sprintf("Writing %s (includes %s tests)", $filename, count($tests)));
             file_put_contents($filename, implode("\n", $tests));
         }
     }
+
+    /**
+     * Determines if the test is part of the filter group.
+     *
+     * @param $test mixed Any kind of codeption test. Cest or Cept
+     * @return bool Returns "true" if the test is part of the filter group, otherwise "false".
+     */
+    protected function matchesToFilterGroup($test)
+    {
+        $isFilterGroupMatched = false;
+
+        if (false === empty($this->filterGroup)) {
+            $phpunitGroups = [];
+
+            if ($test instanceof Cept) {
+                $phpunitGroups = Annotation::fetchAllFromComment(
```

Comment:
can't you get groups with the same way as for Cest?
Using `groupManager->groupsForTest`?